script/copy_deduplicate

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env python3
+
+"""
+Copy a day's data from live ping tables to stable ping tables,
+deduplicating on document_id.
+
+By default, the script will process all tables in datasets named
+like *_live, copying data into table of the same name in *_stable
+datasets. The script can be configured to exclude a list of tables
+or to process only a specific list of tables.
+"""
+
+from argparse import ArgumentParser
+from datetime import datetime
+
+from google.cloud import bigquery
+
+QUERY_TEMPLATE = """\
+WITH
+  numbered_duplicates AS (
+  SELECT
+    *,
+    ROW_NUMBER() OVER (PARTITION BY document_id ORDER BY submission_timestamp) AS _n
+  FROM
+    `{source_table_spec}`
+  WHERE
+    DATE(submission_timestamp) = @submission_date )
+  --
+SELECT
+  * EXCEPT (_n)
+FROM
+  numbered_duplicates
+WHERE
+  _n = 1
+"""
+
+parser = ArgumentParser(description=__doc__)
+parser.add_argument(
+    "--project-id",
+    default="moz-fx-data-shar-nonprod-efed",
+    help="ID of the project in which to run query jobs",
+)
+parser.add_argument(
+    "--date",
+    required=True,
+    type=lambda d: datetime.strptime(d, "%Y-%m-%d").date(),
+    help="Which day's data to copy, in format 2019-01-01",
+)
+parser.add_argument(
+    "--dry-run",
+    action="store_true",
+    help=("Do not run queries, but instead print the query job config "
+          "and bytes that would be processed"),
+)
+group = parser.add_mutually_exclusive_group()
+group.add_argument(
+    "--only",
+    nargs="+",
+    dest="only_tables",
+    default=[],
+    help=("Process only the given tables; "
+          "pass names like 'telemetry_live.main_v4'"),
+)
+group.add_argument(
+    "--except",
+    nargs="+",
+    dest="except_tables",
+    default=[],
```

Comment:
After merging in suggestions, I realize I had a good reason for not giving these a default. For `--only`, having an empty list as default leads to us always rejecting table names, since no table every matches an entry in the empty list.

To keep these consistent and to make intent clear, I'm removing the defaults and doing explicit `is not None` checks for both arguments.